Client/Frontend/Browser/BackForwardListView.swift

```diff
@@ -0,0 +1,196 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+import WebKit
+import Storage
+import SnapKit
+
+class BackForwardListView: UIView, UITableViewDataSource, UITableViewDelegate {
+    
+    enum BackForwardType {
+        case Forward
+        case Current
+        case Backward
+    }
+    
+    struct BackForwardViewUX {
+        static let RowHeight = 50
+    }
+    
+    private var profile: Profile?
+    private var sites = [String: Site]()
+    private var tableView: UITableView!
+    private var isPrivate: Bool!
+    private var dismissing = false
+    var tabManager: TabManager!
```

Comment:
In general try to avoid using optionals as much as you can unless you need them to convey an empty state. For example, since we pass in `profile` in the constructor, that doesn't need to be an optional. Additionally, try to avoid explicit unwrapped optionals. They tend to be a source of exceptions so I find minimizing their use helps a lot. Some solutions include using `lazy` properties or using regular optionals if you're not sure the value will exist.